src/Container.js

```diff
@@ -47,17 +47,25 @@ export default (Pure) => {
   }
 
   function hookActions(actions) {
-    return _.mapValues(actions, (action, key) => (data, options) => {
-      const id = action(data, { delegate: false, ...options });
+    return _.mapValues(actions, (action, key) => (...data) => {
+      const id = action(...data);
       actionStack.push({ id, key });
     });
   }
 
+  function createActionsInstances(actions, options) {
+    return _.mapValues(actions, (action) => {
+      const actionInstance = action(options);
+      return (...args) => actionInstance._call(...args);
+    });
+  }
+
   function wrapDispatchToProps() {
     const dispatchToProps = Pure.dispatchToProps || {};
     return (dispatch, ownProps) => {
       const actionCreators = resolve(dispatchToProps, { dispatch, ownProps });
-      const boundActions = bindActionCreators(actionCreators, dispatch);
+      const actionsInstances = createActionsInstances(actionCreators, { delegate: false });
```

Comment:
Ok ... a bit confusing!
* technically createActionInstance does not return an instance
* also it would be `actionInstances` and not `actionsInstances` (no `s` :) ) 
* We might want to find a way to test whether it returns an actual action instance or a plain `function`. 

In Action.js : 
* Maybe we should rename `_call` since there's no wrapper around it anymore? 
* I am wondering if returning the `id` in call still makes sense in that context ... the id should probaby be bound to the instance

For now something like this could be a bit clearer in my opinion: 
```
  function hookActions(actionConstructors) {
    return _.mapValues(actionConstructors, (actionConstructor, key) => (...data) => {
      const instance = actionConstructor({ delegate: false });
      const id = instance._call(...data)
      actionStack.push({ id, key });
    });
  }

  function wrapDispatchToProps() {
    const dispatchToProps = Pure.dispatchToProps || {};
    return (dispatch, ownProps) => {
      const actionConstructors = resolve(dispatchToProps, { dispatch, ownProps });
      const hookedActions = hookActions(actionConstructors);
      const actions = bindActionCreators(hookedActions, dispatch);
      return { dispatch, ...actions };
    };
  }

```

Reply:
In fact this does not even work, if you dispatch multiple times the same action from a container, you are using the same instance (just encountered the problem with multiple file uploads)
I don't know if we can use the redux bindactioncreators here, we may need to manually pass dispatch somehow.